### PR TITLE
Unify IsPoiChecker and use it for 'operator'.

### PR DESCRIPTION
**Pull request by vng**
Merge after the upcoming release.

### CODE REVIEW

File: indexer/ftypes_matcher.cpp [hunk 1]
```diff
@@ -459,29 +419,51 @@ IsPisteChecker::IsPisteChecker() : BaseChecker(1 /* level */)
...
-    "railway",
-    "highway",
-    "aeroway",
-    "healthcare",
+
+OneLevelPOIChecker::OneLevelPOIChecker() : ftypes::BaseChecker(1 /* level */)
+{
+  Classificator const & c = classif();
+
+  auto paths = {
```
Comment (biodranik) [inline indexer/ftypes_matcher.cpp:427]:
```suggestion
  auto const paths = {
```

File: indexer/ftypes_matcher.cpp [hunk 2]
```diff
@@ -459,29 +419,51 @@ IsPisteChecker::IsPisteChecker() : BaseChecker(1 /* level */)
...
+      "amenity",  "craft", "healthcare", "historic", "leisure", "office", "railway", "shop", "sport", "tourism",
+  };
+  for (auto const & path : paths)
```
Comment (biodranik) [inline indexer/ftypes_matcher.cpp:430]: This reduces the number of asm instructions by two. Unfortunately, compilers are not smart :(
```suggestion
  for (auto const * path : paths)
```

File: indexer/ftypes_matcher.hpp [hunk 3]
```diff
@@ -316,14 +261,35 @@ class IsPisteChecker : public BaseChecker
...
+};
+
+class IsPoiChecker
 {
-  IsPoiChecker();
 public:
   DECLARE_CHECKER_INSTANCE(IsPoiChecker);
+
+  bool operator()(FeatureType & ft) const { return m_oneLevel(ft) || m_twoLevel(ft); }
+  template <class T> bool operator()(T const & t) const { return m_oneLevel(t) || m_twoLevel(t); }
```
Comment (biodranik) [inline indexer/ftypes_matcher.hpp:285]: Are there threeLevel types? Are they POI?

File: generator/osm2meta.cpp [hunk 4]
```diff
@@ -102,20 +102,8 @@ std::string MetadataTagProcessorImpl::ValidateAndFormat_stars(std::string const
...
-      ftypes::IsMoneyExchangeChecker::Instance()(t) ||
-      ftypes::IsFuelStationChecker::Instance()(t) ||
-      ftypes::IsRecyclingCentreChecker::Instance()(t) ||
-      ftypes::IsPostPoiChecker::Instance()(t) ||
-      ftypes::IsCarSharingChecker::Instance()(t) ||
-      ftypes::IsCarRentalChecker::Instance()(t) ||
-      ftypes::IsBicycleRentalChecker::Instance()(t) ||
-      ftypes::IsParkingChecker::Instance()(t))
-  {
+  if (ftypes::IsPoiChecker::Instance()(m_params.m_types))
```
Comment (pastk) [inline generator/osm2meta.cpp:105]: Hmm, maybe there was a reason "operator" was picked up for certain POIs only - probably only for those where it should highly likely make sense.

E.g. a particular restaurant in a franchise chain could be operated by some small/unknown company. See https://wiki.openstreetmap.org/wiki/Key:operator#Examples

Comment (vng) [reply generator/osm2meta.cpp:105]: It is a follow-up from [here](https://github.com/organicmaps/organicmaps/pull/8523) and [here](https://github.com/organicmaps/organicmaps/issues/8039)

You can open a discussion there :)

Comment (pastk) [reply generator/osm2meta.cpp:105]: I did :)

Comment (vng) [reply generator/osm2meta.cpp:105]: I also have mixed feelings about widely used 'operator' and 'network' tags, but POIs look like a reasonable tradeoff.

Comment (biodranik) [reply generator/osm2meta.cpp:105]: What are the typical values in some well-mapped city? Let's see if they're useful for POIs.

**Comment by vng** [comment]
Ok, while the discussion is still open, I made like before plus "university", "vending_machine" (from the issue).

**Comment by pastk** [comment]
The issue mentioned public transport ticket vending machines - makes sense.

Re other vending machines (e.g. drinks) it could be some small unknown companies operating them... But IDK really.
It'd be good if someone makes a tagging research.
Or we can push it to alpha and ask for feedback.

**Review by pastk — APPROVED** [review]
Technically it looks good!

**Comment by pastk** [comment]
> The issue mentioned public transport ticket vending machines - makes sense.
> 
> Re other vending machines (e.g. drinks, coffee) it could be some small unknown companies operating them...

Is it easy to limit the vending machines change to `amenity-vending_machine-public_transport_tickets` only?

**Comment by dvdmrtnz** [comment]
Changes look good, only one comment about this:

> Ok, while the discussion is still open, I made like before (operator selector) plus "university", "vending_machine" (from the issue).

In https://github.com/organicmaps/organicmaps/issues/8039 the user actually wanted to be able to see the operator of buildings operated by a university. So these buildings would not be tagged with `amenity=university` but with `building`. 

I understand that most building's operators are going to be irrelevant for users, so maybe only show operator for university-related buildings like `building=dormitory`, `building=university` ?

**Pull request merged**